### DISCUSSION TITLE
accept only long command line options

### DISCUSSION THREAD

**Discussion by williamh** [linux-speakup/espeakup#42] [Ideas]
Hey all,

is it reasonable for us to transition away from short command line options and accept only long ones in the future?
I think it might be easier to use long options rather than using different case on short options and keeping track of the short options. For example, I'm currently looking at a --default-volume option, but the short option will have to be -o because both -v and -V are used already.

**Comment by zahyur** [comment]
I don't think it is reasonable to get rid of the short options just
because you've had to use something other than the initial letter once.
If someone doesn't like them - they could use the long options.
But someone may not like the long options as well, and could prefer the
short ones.
I'm not sure you should use 'o' - it sounds too much like 'output'.
But on other hand, using the letter on the incremental position if all
previous are already taken sounds the most logical choice. 
Could there be 'output' or anything with 'o' - ever?

**Reply by williamh** [reply]
@zahyur Currently I don't know of anything that we have that would use -o but that doesn't speak for what could happen in the future. Espeak could come up with something that we might want to add that would best map to -o or -O. I'm not sure it is any better for other single letter options.
I may look at omitting the short option in this case and dropping it for some of the other options.

**Comment by alex19EP** [comment]
I think we just can omit short options for some parameters. no need to eliminate them completely.

**Reply by williamh** [reply]
This is a possibility. I'll look into getopt_long behavior and see if I can make that happen.

**Comment by zahyur** [comment]
The main problem with short options is that they can't be changed for
backward compatibility. Otherwise, I think 'u' sounds like 'volume',
too. It's usually 'uninstall', 'upgrade' or something like that.
'l' is usually 'list', 'limit', 'length', so it would be strange.
A function for choosing the output device could exist, so that could be
'o'. The volume could be a capital 'O' in that case.
